test/invalidNumber.js

```diff
@@ -0,0 +1,76 @@
+var JsonParse = require('../jsonparse');
+var test = require('tape');
+
+const {precisionWasLost} = JsonParse.testing
+
+test('determines that precision is lost for an unsafely precise positive integer', t => {
+  const jsonString = '123908471290347091237490182734091237'
+  t.assert(precisionWasLost(jsonString, Number(jsonString)))
+
+  t.end()
+})
+
+test('determines that precision is not lost for a safely precise positive integer', t => {
+  const jsonString = '12390847129'
+  t.assert(!precisionWasLost(jsonString, Number(jsonString)))
+
+  t.end()
+})
+
+test('determines that precision is lost for an unsafely precise negative integer', t => {
+  const jsonString = '-123908471290347091237490182734091237'
+  t.assert(precisionWasLost(jsonString, Number(jsonString)))
+
+  t.end()
+})
+
+test('determines that precision is not lost for a safely precise negative integer', t => {
+  const jsonString = '-123908471290'
+  t.assert(!precisionWasLost(jsonString, Number(jsonString)))
+
+  t.end()
+})
+
+test('determines that precision is lost for an unsafely precise value in decimal notation', t => {
```

Comment:
For completeness, can we add a test that verifies it `'determines that precision is not lost for a safely precise value in decimal notation'`?